cmd/ecrecover/ecrecover.go

```diff
@@ -0,0 +1,133 @@
+package ecrecover
+
+import (
+	_ "embed"
+	"encoding/json"
+	"io"
+	"math/big"
+	"os"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	ethrpc "github.com/ethereum/go-ethereum/rpc"
+	"github.com/maticnetwork/polygon-cli/rpctypes"
+	"github.com/maticnetwork/polygon-cli/util"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+var (
+	//go:embed usage.md
+	usage string
+
+	rpcUrl      string
+	blockNumber uint64
+	filePath    string
+)
+
+var EcRecoverCmd = &cobra.Command{
+	Use:   "ecrecover",
+	Short: "Recovers and returns the public key of the signature",
+	Long:  usage,
+	Args:  cobra.NoArgs,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		return checkFlags()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+
+		var block *types.Block
+		var err error
+
+		if rpcUrl == "" {
+			block = new(types.Block)
+			var blockJSON []byte
+			if filePath != "" {
+				blockJSON, err = os.ReadFile(filePath)
```

Comment:
Can you break up the run command into some functions. I think it's a little too nested and affects readability.